### PR TITLE
Docstrings

### LINKED ISSUE #68: Add docstrings to Ingestor/Analyzer
- [x] Analyzer
- [x] Ingestor

### PR DESCRIPTION
**Pull request by finestructure**
Fixes #68 and tidies up a few small issues (typos etc)

### CODE REVIEW

File: Tests/AppTests/WebpageSnapshotTests.swift [hunk 1]
```diff
@@ -24,6 +24,7 @@ let recordSnapshotForAllTests = false
 class WebpageSnapshotTests: XCTestCase {
     
     override func setUpWithError() throws {
+        try XCTSkipIf((Environment.get("SKIP_SNAPSHOTS") ?? "false") == "true")
```
Comment (finestructure) [inline Tests/AppTests/WebpageSnapshotTests.swift:27]: I've added this so I can easily run the tests locally without the snapshots polluting the results. However, unless you run the tests every once in a while on Catalina we've got a bit of a test gap here. I'll add an issue to see if we can also run the tests on macOS in CI.

Comment (finestructure) [reply Tests/AppTests/WebpageSnapshotTests.swift:27]: #711

Comment (daveverwer) [reply Tests/AppTests/WebpageSnapshotTests.swift:27]: We might get a release date for Big Sur tomorrow, maybe. I'll upgrade as soon as it hits GM.

Comment (daveverwer) [reply Tests/AppTests/WebpageSnapshotTests.swift:27]: I just ran the tests on Catalina with snapshots on this branch and they pass 👍 I'll keep doing that before merging anything until we have this sorted out.

**Review by daveverwer — APPROVED** [review]

**Pull request merged**